modules/opentelemetry/src/main/java/com/spotify/ffwd/opentelemetry/OpenTelemetryPluginSink.java

```diff
@@ -0,0 +1,174 @@
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Metadata;
+import io.grpc.stub.MetadataUtils;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceResponse;
+import io.opentelemetry.proto.collector.metrics.v1.MetricsServiceGrpc;
+import io.opentelemetry.proto.common.v1.StringKeyValue;
+import io.opentelemetry.proto.metrics.v1.DoubleDataPoint;
+import io.opentelemetry.proto.metrics.v1.DoubleGauge;
+import io.opentelemetry.proto.metrics.v1.InstrumentationLibraryMetrics;
+import io.opentelemetry.proto.metrics.v1.ResourceMetrics;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OpenTelemetryPluginSink implements PluginSink {
+    private static final Logger log = LoggerFactory.getLogger(OpenTelemetryPluginSink.class);
+
+    @Inject private AsyncFramework async;
+    private String endpoint;
+    private Map<String, String> headers;
+    private ManagedChannel channel;
+    private MetricsServiceGrpc.MetricsServiceBlockingStub stub;
+
+    OpenTelemetryPluginSink(
+        String endpoint,
+        Map<String, String> headers
+    ) {
+        this.endpoint = endpoint;
+        this.headers = headers;
+    }
+
+
+    @Override
+    public void sendMetric(Metric metric) {
+        if (metric.hasDistribution()) {
+            log.warn("Distributions not supported, dropping metric");
+            return;
+        }
+
+        Map<String, String> tags = metric.getTags();
+
+        // NB(hexedpackets): Name is supposed to uniquely identify the metric. The combination
+        // of `key` and `what` may or may not be adequate. It might make sense to have this be
+        // configurable?
+        String name = metric.getKey() + "." + tags.get("what");
+        DoubleGauge gauge = convertGauge(metric);
+        io.opentelemetry.proto.metrics.v1.Metric.Builder metricBuilder =
+            io.opentelemetry.proto.metrics.v1.Metric.newBuilder()
+                .setName(name)
+                .setDoubleGauge(gauge);
+
+        String unit = tags.get("unit");
+        if (unit != null) {
+            metricBuilder.setUnit(unit);
+        }
+
+        ExportMetricsServiceRequest request = ExportMetricsServiceRequest.newBuilder()
+            .addResourceMetrics(ResourceMetrics.newBuilder()
+                .addInstrumentationLibraryMetrics(
+                    InstrumentationLibraryMetrics.newBuilder().addMetrics(metricBuilder.build())
+                        .build())
+                .build())
+            .build();
+
+        try {
+            //noinspection ResultOfMethodCallIgnored
+            stub.export(request);
+        } catch (Exception e) {
+            log.error(e.getMessage());
+        }
+    }
+
+    @Override
+    public void sendBatch(Batch batch) {
```

Comment:
I wonder why batch is not supported ?

Reply:
Its just not necessary right now, and the best way to map a batch to the OT metric definition isn't 100% clear so I'm kicking the can for the moment.